airflow/providers/databricks/sensors/databricks_table_changes.py

```diff
@@ -0,0 +1,164 @@
+    :param caller: String passed to name a hook to Databricks, defaults to "DatabricksPartitionSensor"
+    :param client_parameters: Additional parameters internal to Databricks SQL Connector parameters.
+    :param timestamp: Timestamp to check event history for a Delta table,
+        defaults to datetime.now()-timedelta(days=7)
+    """
+
+    template_fields: Sequence[str] = ("databricks_conn_id", "catalog", "schema", "table_name", "timestamp")
+
+    def __init__(
+        self,
+        *,
+        databricks_conn_id: str = DatabricksSqlHook.default_conn_name,
+        http_path: str | None = None,
+        sql_endpoint_name: str | None = None,
+        session_configuration=None,
+        http_headers: list[tuple[str, str]] | None = None,
+        catalog: str = "",
+        schema: str = "default",
+        table_name: str = "",
+        handler: Callable[[Any], Any] = fetch_all_handler,
+        timestamp: datetime = datetime.now() - timedelta(days=7),
+        caller: str = "DatabricksTableChangesSensor",
+        client_parameters: dict[str, Any] | None = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.databricks_conn_id = databricks_conn_id
+        self._http_path = http_path
+        self._sql_endpoint_name = sql_endpoint_name
+        self.session_config = session_configuration
+        self.http_headers = http_headers
+        self.catalog = catalog
+        self.schema = schema
+        self.table_name = table_name
+        self.timestamp = timestamp
+        self.caller = caller
+        self.client_parameters = client_parameters or {}
+        self.hook_params = kwargs.pop("hook_params", {})
+        self.handler = handler
+
+    def _get_hook(self) -> DatabricksSqlHook:
+        return DatabricksSqlHook(
+            self.databricks_conn_id,
+            self._http_path,
+            self._sql_endpoint_name,
+            self.session_config,
+            self.http_headers,
+            self.catalog,
+            self.schema,
+            self.caller,
+            **self.client_parameters,
+            **self.hook_params,
+        )
+
+    def _sql_sensor(self, sql):
+        hook = self._get_hook()
+        sql_result = hook.run(
+            sql,
+            handler=self.handler if self.do_xcom_push else None,
+        )
+        return sql_result
+
+    @staticmethod
+    def get_previous_version(context: Context, lookup_key):
+        return context["ti"].xcom_pull(key=lookup_key, include_prior_dates=True)
+
+    @staticmethod
+    def set_version(context: Context, lookup_key, version):
+        context["ti"].xcom_push(key=lookup_key, value=version)
+
+    def get_current_table_version(self, table_name, time_range):
+        change_sql = (
+            f"SELECT COUNT(version) as versions from "
+            f"(DESCRIBE HISTORY {table_name}) "
+            f"WHERE timestamp >= '{time_range}'"
```

Comment:
if time_range doesn't change between calls, we may return true all the time. Why not compare with the latest version instead?

Reply:
Does it make sense to report only data changes? Otherwise we'll get not necessary changes like after `VACUUM`, `OPTIMIZE`, ...

Reply:
Added these changes in the recent push.